File: travelAgency/src/main/java/group/travelagency/repository/FlightRepository.java
package group.travelagency.repository;

import group.travelagency.domain.Flight;
import group.travelagency.domain.Location;
import group.travelagency.domain.dto.FlightDto;
import group.travelagency.repository.database.FlightTable;
import group.travelagency.repository.database.Table;
import group.travelagency.utils.Constants;
import group.travelagency.utils.Resources;

import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.util.stream.Collectors;

public class FlightRepository implements Repository<Long, Flight> {
    private final Table<Long, FlightDto> table;

    public FlightRepository() throws SQLException {
        table = (FlightTable) Resources.getTableFactory().getTable(Constants.Db.Tables.FLIGHT);
    }

    @Override
    public void add(Flight flight) {
        FlightDto flightDto = new FlightDto(flight.getStart().getId(), flight.getDestination().getId(), flight.getStartDate(),
                flight.getNrOfSeats());
        flightDto.setId(flight.getId());
        table.add(flightDto);
    }

    @Override
    public void update(Flight entity) {
        //TODO
    }

    @Override
    public Optional<Flight> findByID(Long aLong) throws SQLException {
        Optional<FlightDto> optionalFlightDto = table.findById(aLong);
        if (optionalFlightDto.isPresent()) {
            Optional<Location> start = Resources.getInstance().getLocationRepository().findByID(optionalFlightDto.get().getStartId());
            Optional<Location> destination = Resources.getInstance().getLocationRepository().findByID(optionalFlightDto.get().getDestinationId());
            if (start.isPresent() && destination.isPresent()) {
                Flight flight = new Flight(start.get(), destination.get(),
                        optionalFlightDto.get().getStartDate(), optionalFlightDto.get().getNrOfSeats());
                return Optional.of(flight);
            }
        }
        return Optional.empty();
    }

    @Override
    public void delete(Long aLong) {

    }

    @Override
    public List<Flight> getAll() {
        List<FlightDto> flightDtos = table.getAll();
        List<Flight> flights = new ArrayList<>();
        for (FlightDto flightDto : flightDtos) {
            try {
                Optional<Location> start = Resources.getInstance().getLocationRepository().findByID(flightDto.getStartId());
                Optional<Location> destination = Resources.getInstance().getLocationRepository().findByID(flightDto.getDestinationId());
                if (start.isPresent() && destination.isPresent()) {
                    flights.add(new Flight(start.get(), destination.get(),
                            flightDto.getStartDate(), flightDto.getNrOfSeats())
                    );
                }
            } catch (SQLException throwables) {
                throwables.printStackTrace();
            }
        }
        return flights;
    }
}

File: travelAgency2/travelAgency2/Repository/FlightRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using travelAgency2.Domain.Dto;
using travelAgency2.Repository.Database;
using travelAgency2.src.Domain;
using travelAgency2.src.Repository;
using travelAgency2.Utils;

namespace travelAgency2.Repository
{
    internal class FlightRepository : Repository<long, Flight>
    {
        private Table<long, FlightDto> table;

        public FlightRepository()
        {
            table = (FlightTable) Resources.getTableFactory().getTable(Constants.Db.Tables.FLIGHT);
        }

        public void add(Flight flight)
        {
            FlightDto flightDto = new FlightDto(flight.Id, flight.start.Id, flight.destination.Id, flight.startDate,
                flight.nrOfSeats);
            table.add(flightDto);
        }

        public void delete(long id)
        {
            throw new NotImplementedException();
        }

        public Flight findByID(long id)
        {
            FlightDto flightDto = table.findById(id);
            Location start = Resources.getInstance().getLocationRepository().findByID(flightDto.startId);
            Location destination = Resources.getInstance().getLocationRepository().findByID(flightDto.destinationId);
            return new Flight(flightDto.Id, start, destination, flightDto.startDate, flightDto.nrOfSeats);
        }

        public List<Flight> getAll()
        {
            List<FlightDto> flightDtos = table.getAll();
            List<Flight> flights = new List<Flight>();
            foreach (FlightDto flightDto in flightDtos)
            {
                Location start = Resources.getInstance().getLocationRepository().findByID(flightDto.startId);
                Location destination = Resources.getInstance().getLocationRepository().findByID(flightDto.destinationId);
                flights.Add(new Flight(flightDto.Id, start, destination, flightDto.startDate, flightDto.nrOfSeats));
            }
            return flights;
        }

        public void update(Flight entity)
        {
            throw new NotImplementedException();
        }
    }
}
